card/mod/01_history/set/all/actions.rb

```diff
@@ -0,0 +1,47 @@
+# -*- encoding : utf-8 -*-
```

Comment:
good thoughts.

I didn't mean to suggest they weren't related.  If we're talking *design*, then it makes sense to design them together.  But if we're talking implementation, it's best to bite off smaller chunks.